src/backend/core/tests/services/test_tiered_storage.py

```
@@ -0,0 +1,726 @@
+    def test_encrypt_with_missing_active_key(self):
+        """Test that encrypt fails if active_key_id not in encryption_keys."""
+        service = TieredStorageService()
+        service.encryption_keys = {"1": Fernet.generate_key().decode()}
+        service.active_key_id = 99  # Not in keys
+
+        with pytest.raises(ValueError, match="Active encryption key_id 99 not found"):
+            service.encrypt(b"test data")
+
+    def test_decrypt_with_corrupted_data(self):
+        """Test that decrypt fails gracefully with corrupted Fernet data."""
+        from cryptography.fernet import InvalidToken
```

Ruff (0.14.14)

`import` should be at the top-level of a file

(PLC0415)